_source/_posts/2021-12-06-spring-data-elasticsearch.md

```diff
@@ -0,0 +1,338 @@
+
+```yml
+services:
+...  
+  blog-kibana:
+     image: docker.elastic.co/kibana/kibana:7.13.3
+     ports:
+       - 5601:5601
+     environment:
+       ELASTICSEARCH_URL: http://blog-elasticsearch:9200
+       ELASTICSEARCH_HOSTS: '["http://blog-elasticsearch:9200"]'
+...       
+```
+
+Also add the following additional configuration to the `blog-elasticsearch` service, to expose the ports to the local host for sending APIs requests:
+
+```yml
+services:
+  blog-elasticsearch:
+    ...
+    ports:
+      - 9200:9200
+```
+Remove the Keycloak service at the bottom, as Okta will be used as Identity Provider.
+
+Create each applications' container image:
+
+```shell
+cd blog
+./mvnw -DskipTests -ntp -Pprod verify jib:dockerBuild
+```
+
+```shell
+cd gateway
+./mvnw -DskipTests -ntp -Pprod verify jib:dockerBuild
+```
+
+Run the architecture with Docker Compose:
+
+```shell
+cd ../docker-compose
+docker compose up
+```
+ Access the JHipster Registry at [**http://localhost:8761**](http://localhost:8761) and sign in with your Okta credentials. When you see all services up and green, go to http://localhost:8080 and sign in with your Okta account.
+
+ {% img blog/spring-data-elasticsearch/okta-signin.png alt:"Okta Sign In Form" width:"400" %}{: .center-image }
+
+
+ **IMPORTANT NOTE**: There is a potential issue observed for the blog service, as [Kubernetes probes](https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.endpoints.kubernetes-probes) are enabled by default, the blog application might change its state to `OUT_OF_SERVICE` during start up and Eureka discards the following heartbeats with state `UP`. In that case, disable Kubernetes probes in the `docker-compose.yml` with the following environment variable:
+
+```yml
+services:
+  blog:
+    image: blog
+    environment:
+    ...
+      - MANAGEMENT_ENDPOINT_HEALTH_PROBES_ENABLED=false
+    ...    
+```
+
+## Inspecting Elasticsearch index mapping
+
+During start-up, Spring Data Elasticsearch will create the index for the entities annotated with `@Document`, deriving the mappings from the entity's annotations. But for the properties, if the field type is not specified, it defaults to FieldType.Auto. This means, that no mapping entry is written for the property and that Elasticsearch will add a mapping entry dynamically when the first data for this property is stored.
+
+As Kibana interface was configured in Docker Compose, before creating any entities, let's inspect the index mappings that were automatically created for the `blog` microservice. Go to [**http://localhost:5601**](http://localhost:5601), the Kibana dashboard. On the top left menu, go to **Management** > **Stack Management** > **Index Management** > **Indices** tab. Besides the `user` index, an index per entity should be listed.
+
+{% img blog/spring-data-elasticsearch/kibana-indexes.png alt:"Kibana Indexes" width:"800" %}{: .center-image }
```

Comment:
When I first go to `http://localhost:5601`, I see the following screen.

<img width="656" alt="Screen Shot 2022-01-26 at 15 25 19" src="https://user-images.githubusercontent.com/17892/151257470-beed7d50-b038-4e74-b4e3-6f55ff66b5cc.png">

If I go to the Indices tab, I see "Error loading indices". 

<img width="1323" alt="Screen Shot 2022-01-26 at 15 27 22" src="https://user-images.githubusercontent.com/17892/151257709-2eaf9d7c-974c-470d-986c-25402af8f29a.png">

I'm using JHipster 7.6.0. Do you think this could be causing the issue? If so, we should file an issue and get it fixed.

Reply:
This was caused by the Kibana vs Elasticsearch version mismatch.